cmd/server-mux.go

```diff
@@ -110,28 +103,21 @@ func (c *ConnMux) PeekProtocol() string {
 // Read - streams the ConnMux buffer when reset flag is activated, otherwise
 // streams from the incoming network connection
 func (c *ConnMux) Read(b []byte) (int, error) {
-	if c.dataBuf.unRead {
-		n := copy(b, c.dataBuf.buffer[c.dataBuf.offset:])
-		c.dataBuf.offset += n
-		if c.dataBuf.offset == len(c.dataBuf.buffer) {
-			// We finished copying all c.buffer, reset all
-			c.dataBuf.unRead = false
-			c.dataBuf.offset = 0
-			c.dataBuf.buffer = c.dataBuf.buffer[:]
-			if n < len(b) {
-				// Continue copying from socket if b still has room for data
-				tmpBuffer := make([]byte, len(b)-n-1)
-				nr, err := c.Conn.Read(tmpBuffer)
-				for idx, val := range tmpBuffer {
-					b[n+idx] = val
-				}
-				return n + nr, err
-			}
-		}
-		// We here return the last error
-		return n, c.lastError
+	if c.lastError != nil {
```

Comment:
`lastError` field can be removed from ConnMux as c.bufwr.Read(b) below will return if any.